// Copyright 2019 Shift Cryptosecurity AG
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package shiftcrypto.bitbox02;

message PubResponse {
  string pub = 1;
}

message RootFingerprintRequest {
  bool display = 1;
}

message RootFingerprintResponse {
  bytes fingerprint = 1;
}

// See https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki.
// version field dropped as it will set dynamically based on the context (xpub, ypub, etc.).
message XPub {
  bytes depth = 1;
  bytes parent_fingerprint = 2;
  uint32 child_num = 3;
  bytes chain_code = 4;
  bytes public_key = 5;
}

// This message exists for use in oneof or repeated fields, where one can't inline `repeated uint32` due to protobuf rules.
message Keypath {
  repeated uint32 keypath = 1;
}
